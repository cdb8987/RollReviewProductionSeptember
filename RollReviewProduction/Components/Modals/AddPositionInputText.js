
import React from 'react'
import {useState} from 'react'
import { Button, View}from 'react-native'
import {TextInput} from 'react-native-paper'


export default function AddPositionInputText({handleAdd,positions, setPositions, onDismiss }){

        const [text, setText] = useState('')
        

        return (
            <>
            <TextInput
                style={{height: 40, flex: 1, backgroundColor: 'white'}}
                placeholder="..."
                onChangeText={(newText) => {setText(newText)}}
                defaultValue={text}
                outlined={true}
                />
            <Button 
                style={{marginTop: 15}}
                compact={true} 
                mode={'contained'} 
                onPress={()=>{
                    const newItem = text.charAt(0).toUpperCase()+text.slice(1)
                    if(newItem && !positions.includes(newItem)){
                    handleAdd(positions, setPositions, newItem)
                    onDismiss()}
                    
                    else{ onDismiss()}
                }}
                title={'ADD'}
                
            ></Button>
            </>

        )

        
}



