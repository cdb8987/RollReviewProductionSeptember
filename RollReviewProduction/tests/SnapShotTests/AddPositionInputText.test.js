import React from "react";
import {act} from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import AddPositionInputText from '../../Components/Modals/AddPositionInputText.js'
import { Button}from 'react-native'
import {TextInput} from 'react-native-paper'



test('renders AddPositionInputText snapshot',  () => {
    const handleAdd = jest.fn()
    const setPositions = jest.fn()
    const onDismiss = jest.fn()
    
    const renderer = new ShallowRenderer()
    const tree = renderer.render(<AddPositionInputText 
        handleAdd={handleAdd}
        positions={["Back Control", "Durkaposition", "Guard", "Mount", "Neweventposition", "Newposition", "Newposition2", "Newposition3", "Newposition4", "Newposition5", "Newposition6", "Newrollposition", "Newtutorialposition", "Productionposition", "Side Control", "Testytreest"]}
        setPositions={setPositions}
        onDismiss={onDismiss}
    />);

    expect(tree).toMatchSnapshot();
    
});