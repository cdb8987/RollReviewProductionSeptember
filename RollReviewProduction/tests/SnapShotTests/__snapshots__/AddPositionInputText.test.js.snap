// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddPositionInputText snapshot 1`] = `
<React.Fragment>
  <ForwardRef
    defaultValue=""
    onChangeText={[Function]}
    outlined={true}
    placeholder="..."
    style={
      {
        "backgroundColor": "white",
        "flex": 1,
        "height": 40,
      }
    }
  />
  <Button
    compact={true}
    mode="contained"
    onPress={[Function]}
    style={
      {
        "marginTop": 15,
      }
    }
    title="ADD"
  />
</React.Fragment>
`;
